### PR TITLE
[DEV-1299] Handle user update attribute event in CognitoFunctions

### PR DESCRIPTION
**Pull request by kin0992**
Handle the event triggered when the user updates the email

#### List of Changes

- Create skelethon of email template
- Add new event to handle

#### Motivation and Context

Before this PR, when the user updates the email address, a verification code was sent to its new email address.
The behaviour we want is that the user receive a verification link.

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/cognito-functions/src/__tests__/custom-message-handler.test.ts [hunk 1]
```diff
@@ -109,4 +110,18 @@ describe('Handler', () => {
       expect(error).toStrictEqual(new Error('Operation not permitted'));
     }
   });
+
+  it('should reply with verification link on user update attributes event', async () => {
```
Comment (kin0992) [inline apps/cognito-functions/src/__tests__/custom-message-handler.test.ts:114]:
```suggestion
  it('should reply with verification link on UpdateUserAttribute event', async () => {
```

File: apps/cognito-functions/src/__tests__/custom-message-handler.test.ts [hunk 2]
```diff
@@ -109,4 +110,18 @@ describe('Handler', () => {
...
+      ...makeEvent(),
+      triggerSource: 'CustomMessage_UpdateUserAttribute',
+    };
+    const { response } = await makeHandler(env)(resendCodeEvent);
+    const { userAttributes, codeParameter } = resendCodeEvent.request;
+    const expected = makeConfirmationUpdateEmailAddress(
+      `https://${env.domain}/auth/email-confirmation?username=${userAttributes['sub']}&code=${codeParameter}`,
+      env.domain
+    );
+    expect(response.emailMessage).toStrictEqual(expected);
```
Comment (kin0992) [inline apps/cognito-functions/src/__tests__/custom-message-handler.test.ts:125]:
```suggestion
    const updateUserAttributeEvent: CustomMessageTriggerEvent = {
      ...makeEvent(),
      triggerSource: 'CustomMessage_UpdateUserAttribute',
    };
    const { response } = await makeHandler(env)(updateUserAttributeEvent);
    const { userAttributes, codeParameter } = updateUserAttributeEvent.request;
    const expected = makeConfirmationUpdateEmailAddress(
      `https://${env.domain}/auth/email-confirmation?username=${userAttributes['sub']}&code=${codeParameter}`,
      env.domain
    );
    expect(response.emailMessage).toStrictEqual(expected);
```

File: apps/cognito-functions/src/templates/confirmation-update-email-address.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    'Ricevi questa e-mail perché hai creato un account su PagoPA DevPortal.<br>Non sei tu? Ignora o cancella questa e-mail.',
+  buttonFallbackText: 'Il bottone non funziona? Puoi usare il seguente link:',
+};
+
+export const makeConfirmationUpdateEmailAddress = (
+  confirmationLink: string,
+  domain: string
+) => parseMjmlToHtml(updateEmailAddressMessage(confirmationLink, domain));
+
+// TODO: This template is just a copy of another one. If something is different, change it here and then remove this comment
```
Comment (datalek) [inline apps/cognito-functions/src/templates/confirmation-update-email-address.ts]:
```suggestion
```

File: apps/cognito-functions/src/templates/confirmation-update-email-address.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import { parseMjmlToHtml } from './mjmlParser';
+
+// FIXME: Change and/or remove some translations. These are just a placeholder
```
Comment (kin0992) [inline apps/cognito-functions/src/templates/confirmation-update-email-address.ts:3]:
```suggestion
```

File: apps/cognito-functions/src/templates/confirmation-update-email-address.ts (hunk 3 above)
Comment (kin0992) [inline apps/cognito-functions/src/templates/confirmation-update-email-address.ts:21]:
```suggestion
```

**Review by datalek — APPROVED** [review]
LGTM

File: apps/cognito-functions/src/templates/confirmation-update-email-address.ts
Comment (kin0992) [inline apps/cognito-functions/src/templates/confirmation-update-email-address.ts:1]: @marcobottaro we are going to merge this PR soon. When the merge is done, you can proceed with the activity https://pagopa.atlassian.net/browse/DEV-1298 and you should update the template in this file.

**Pull request merged**